src/gather_nd.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+import {Tensor, sizeOfShape} from './lib/tensor.js';
+import {validateGatherNDParams} from './lib/validate-input.js';
+
+/**
+ * Gathers values using multidimensional indices.
+ * @param {Tensor} input
+ * @param {Tensor} indices
+ * @return {Tensor}
+ */
+export function gatherND(input, indices) {
+  validateGatherNDParams(input, indices);
+
+  const inputRank = input.rank;
+  const inputShape = input.shape;
+  const indicesRank = indices.rank;
+  const indicesShape = indices.shape;
+  const lastIndicesSize = indicesShape[indicesRank - 1];
+  const tmpShape = inputShape.slice(lastIndicesSize);
+
+  // Refer to https://docs.openvino.ai/2024/documentation/openvino-ir-format/operation-sets/ +
+  //   operation-specs/movement/gather-nd-8.html
```

Comment:
Nit: please keep the URL in one line.

Reply:
Updated in new commits. Please take another look, thanks!